File: C files/Infix,Postfix,Prefix/infixToPostfix.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include "StackA.h"
int precedence(char ch)
{
    switch(ch)
    {
        case '^': return 3;
        case '/': return 2;
        case '*': return 2;
        case '+': return 1;
        case '-': return 1;
        default: return 0;
    }
}
bool is_operator(char ch)
{
    return ch == '^' || ch == '/' || ch=='*' || ch=='+' 
                || ch=='-' || ch=='(' || ch==')';
}
void infix_to_postfix(char infix[], char postfix[])
{
    int k = 0;
    Stack* st = createStack();
    for(int i=0;infix[i]!='\0';i++)
    {
        if(!is_operator(infix[i]))
            postfix[k++] = infix[i];
        else
        {
            if(infix[i]=='(')
                push(st,infix[i]);
            else if(infix[i]==')')
            {
                while(peek(st)!='(')
                    postfix[k++] = pop(st);
                pop(st);
            }
            else
            {
                while(!isEmpty(st) && precedence(infix[i]) <= precedence(peek(st)))
                    postfix[k++] = pop(st);
                push(st,infix[i]);
            }
        }
    }
    while(!isEmpty(st))
        postfix[k++] = pop(st);
    postfix[k] = '\0';
}
int main()
{
    char infix[1000],postfix[1000];
    scanf("%s",infix);
    infix_to_postfix(infix,postfix);
    printf("%s\n",postfix);

}

File: Python files/algorithms/infix,postfix,prefix/infix to postfix.py
def precedence(ch):
    if ch == "^":
        return 3
    elif ch in {"/", "*"}:
        return 2
    elif ch in {"+", "-"}:
        return 1
    else:
        return 0


def is_operator(ch):
    return ch in {"^", "/", "*", "+", "-", "(", ")"}


def infix_to_postfix(infix):
    postfix = []
    stack = []

    for ch in infix:
        if not is_operator(ch):
            postfix.append(ch)
        else:
            if ch == "(":
                stack.append(ch)
            elif ch == ")":
                while stack and stack[-1] != "(":
                    postfix.append(stack.pop())
                stack.pop()
            else:
                while (
                    stack
                    and stack[-1] != "("
                    and precedence(ch) <= precedence(stack[-1])
                ):
                    postfix.append(stack.pop())
                stack.append(ch)

    while stack:
        postfix.append(stack.pop())

    return "".join(postfix)
